subprojects/diagnostics/src/integTest/groovy/org/gradle/api/tasks/diagnostics/DependencyInsightReportVariantDetailsIntegrationTest.groovy

```diff
@@ -78,6 +76,241 @@ project :$expectedProject
+                'org.gradle.jvm.environment': of('', 'standard-jvm'),
+                'org.gradle.jvm.version': of('', JavaVersion.current().majorVersion),
+                'org.gradle.libraryelements': of('', 'classes'),
+                'org.gradle.usage': of('', usage),
+            ])}
+
+${variantOf('runtimeElements', [
+                artifactType: of('jar', ''),
+                'org.gradle.category': of('library', 'library'),
+                'org.gradle.dependency.bundling': of('external', 'external'),
+                'org.gradle.jvm.version': of(JavaVersion.current().majorVersion, JavaVersion.current().majorVersion),
+                'org.gradle.libraryelements': of('jar', 'classes'),
+                'org.gradle.usage': of('java-runtime', usage),
+                'org.gradle.jvm.environment': of('', 'standard-jvm')
+            ])}
+
+${variantOf('runtimeElements-classes', [
+                'org.gradle.category': of('library', 'library'),
+                'org.gradle.dependency.bundling': of('external', 'external'),
+                'org.gradle.jvm.version': of(JavaVersion.current().majorVersion, JavaVersion.current().majorVersion),
+                'org.gradle.libraryelements': of('classes', 'classes'),
+                'org.gradle.usage': of('java-runtime', usage),
+                'org.gradle.jvm.environment': of('', 'standard-jvm')
+            ])}
+
+${variantOf('runtimeElements-resources', [
+                'org.gradle.category': of('library', 'library'),
+                'org.gradle.dependency.bundling': of('external', 'external'),
+                'org.gradle.jvm.version': of(JavaVersion.current().majorVersion, JavaVersion.current().majorVersion),
+                'org.gradle.libraryelements': of('resources', 'classes'),
+                'org.gradle.usage': of('java-runtime', usage),
+                'org.gradle.jvm.environment': of('', 'standard-jvm')
+            ])}
+
+${variantOf('testResultsElementsForTest', [
+                'org.gradle.testsuite.name': of('test', ''),
+                'org.gradle.testsuite.target.name': of('test', ''),
+                'org.gradle.testsuite.type': of('unit-test', ''),
+                'org.gradle.verificationtype': of('test-results', ''),
+                'org.gradle.category': of('verification', 'library'),
+                'org.gradle.dependency.bundling': of('', 'external'),
+                'org.gradle.jvm.environment': of('', 'standard-jvm'),
+                'org.gradle.jvm.version': of('', JavaVersion.current().majorVersion),
+                'org.gradle.libraryelements': of('', 'classes'),
+                'org.gradle.usage': of('', usage),
+            ])}
+
+
+project :$expectedProject
+\\--- $configuration
+"""
+        }
+
+        where:
+        configuration      | expectedVariant   | usage          | requestedLibraryElements
+        'compileClasspath' | 'apiElements'     | 'java-api'     | 'classes'
```

Comment:
I don't see the point of using the parameterization here.

Reply:
Agreed, this was a holdover from a previous implementation. I will remove.